src/main/java/htsjdk/variant/variantcontext/writer/BCF2Writer.java

```diff
@@ -122,6 +122,12 @@
     private VCFHeader lastVCFHeaderOfUnparsedGenotypes = null;
     private boolean canPassOnUnparsedGenotypeDataForLastVCFHeader = false;
 
+    // is the header written to the output stream?
+    private boolean isHeaderWritten;
+
+    // is at least one part of body written to the output stream?
+    private boolean isBodyWritten;
+
```

Comment:
I thought of providing two distinct error messages based on which one (header/body) is written (In our use case, for example, we don't write the header). But I can merge the two variables if it's not important.